tasmota/tasmota_xnrg_energy/xnrg_25_mk_sky_blu.ino

```diff
@@ -0,0 +1,783 @@
+  "6 Batt.UVP connload [V]",   //  ~  Battery UnderVoltageProtection, >=limit reconnects load, typ D03 + 0.2V
+  "7 Comm. address (?)",       //  -  Communication Address (ro)
+  "8 Batt.Type [0=SLA, 1=Li]", // D04 Battery Type: value 0=SLA, 1=LiPo (ro)
+  "9 Batt.System [n*12V]"      //  -  Battery System: 1...4 * 12V (Read-Only, set at Batt.connection)
+};
+
+//  const char MKSB_HTTP_SNS_sS_m_De[]       PROGMEM = "{s}%s " "{m}" " %d{e}";
+//  const char MKSB_HTTP_SNS_str_m_str[]       PROGMEM = "{s}%s " "{m}" " %s{e}";
+//  const char MKSB_HTTP_SNS_str_int_str[]        PROGMEM = "{s}%s " "%d{m}" "%s{e}";
+  const char MKSB_HTTP_SNS_str_m_int[]          PROGMEM = "{s}%s " "{m}" "%d{e}";
+
+/*
+const char * mksb_status_bits[] = {
+                                    // mode
+  "",                               // b0 
+  "",                               // b1
+  D_CHARGE,                         // b2 MPPT Mode Active
+  "",                               // b3
+
+  "",                               // b4
+  "",                               // b5
+  "",                               // b6
+  "",                               // b7
+
+                                            // error
+  D_BATTERY " " D_VOLTAGE " " D_STR_LOW,    // b0 Battery undervoltage (< register 5 / 6)
+  D_BATTERY " " D_VOLTAGE " " D_STR_HIGH,   // b1 Battery overvoltage (> register 1)
+  "",                                       // b2
+  "",                                       // b3
+
+  "",                                       // b4
+  "",                                       // b5
+  "",                                       // b6
+  ""                                        // b7
+};
+*/
+
+#ifdef _WITH_PHASE_NAME_SUPPORT
+const char * mksb_channels[] = {D_DEVICE_INPUT , D_BATTERY};
+#endif
+
+#ifdef _WITH_SERIAL_DEBUGGING
+const char * mksb_rx_counter_names[] = {
+  "Good",
+  "Timeout",
+  "TooShort",
+  "Overflow",
+  "CRC",
+  "Command",
+  "Length"
+};
+#endif
+
+// module data
+mksb_data_t me = { .Serial = nullptr, .pRxBuffer = nullptr }; // singleton
+//mksb_data_t * pMe = nullptr;
+
+
+
+uint8_t MksbChecksum(uint8_t *data, uint8_t len)
+{
+  uint8_t checksum = 0;
+  uint32_t i;
+
+  len -= 1;
+  for (i = 1; i < len; i++) { checksum += data[i]; }
+  return checksum;
+}
+
+
+unsigned long MksbExtractUint(char *data, uint8_t offset_lsb, uint8_t offset_msb)
+{
+	unsigned long result = 0;
+
+  for ( ; offset_msb >= offset_lsb; offset_msb-- ) {
+    result = (result << 8) | (uint8_t)data[offset_msb];
+  }
+	return result;
+}
+
+
+#if 0 // notused
+void MksbSetUint(unsigned long value, uint8_t *data, uint8_t offset_lsb, uint8_t offset_msb)
+{
+    for ( ; offset_lsb <= offset_msb; offset_lsb++ ) {
+      data[offset_lsb] = value & 0xFF;
+      value >>= 8;
+    }
+}
+#endif
+
+
+/********************************************************************************************/
+/* SEND DATA */
+
+bool MksbSend(uint8_t *data, uint8_t len)
```

Comment:
Use `const uint8_t *data` since the data is not changed.

Signature shoudl be
`bool MksbSend(const uint8_t *data, uint8_t len)`